lib/nuget-parser/index.ts

```diff
@@ -130,60 +127,69 @@ Will attempt to build dependency graph anyway, but the operation might fail.`);
       );
     }
   }
+  // If a specific targetFramework has been requested, only query that, otherwise try to do them all
+  const decidedTargetFrameworks = targetFramework
+    ? [targetFramework]
+    : targetFrameworks.filter((framework) => {
+        if (!depsParser.isSupportedByV2GraphGeneration(framework)) {
+          console.log(
+            `\x1b[33m⚠ WARNING\x1b[0m: The runtime resolution flag is currently only supported for the following TargetFrameworks: .NET versions 5 and higher, all versions of .NET Core and all versions of .NET Standard. Detected a TargetFramework: \x1b[1m${framework}\x1b[0m, which will be skipped.`,
+          );
+          return false;
+        }
+        return true;
+      });
 
-  // TODO: OSM-347 - return vulnerability scans for all
-  let decidedTargetFramework: string;
-  if (!targetFramework) {
-    console.log(`No targetFramework supplied, defaulting to using the first in the list of the manifest (\x1b[1m${targetFrameworks[0]}\x1b[0m).
-Supply a targetFramework by using the \x1b[1m--dotnet-target-framework\x1b[0m argument.`);
-    decidedTargetFramework = targetFrameworks[0];
-  } else {
-    decidedTargetFramework = targetFramework;
+  if (decidedTargetFrameworks.length == 0) {
+    throw new InvalidManifestError(
+      `Was not able to find any supported TargetFrameworks to scan, aborting`,
+    );
   }
 
-  if (!depsParser.isSupportedByV2GraphGeneration(decidedTargetFramework)) {
-    throw new FileNotProcessableError(
-      `runtime resolution flag is currently only supported for: .NET versions 5 and higher, all versions of .NET Core and all versions of .NET Standard projects. Supplied versions was parsed as: ${decidedTargetFramework}.`,
+  const results: DotnetCoreV2Results = [];
```

Comment:
Confused diff. I put everything into a `for` loop, so everything is shifted right. But that is all this block did. Everything else is the same.